### PR TITLE
Make datamodel_connector independent from dml 

### PR DESCRIPTION
**Pull request by tomhoule**
Towards a linear parsing pipeline.

AST → ParserDatabase → (core + connector) → Optionally DML

### CODE REVIEW

File: libs/datamodel/core/src/transform/ast_to_dml/lift.rs [hunk 1]
```diff
@@ -469,3 +469,66 @@ fn parser_database_referential_action_to_dml_referential_action(
 fn parser_database_scalar_type_to_dml_scalar_type(st: parser_database::ScalarType) -> dml::ScalarType {
     st.as_str().parse().unwrap()
 }
+
+fn datamodel_connector_native_type_to_dml_native_type(
```
Comment (tomhoule) [inline libs/datamodel/core/src/transform/ast_to_dml/lift.rs:473]: This just moved, no changes

File: libs/datamodel/core/src/transform/dml_to_ast/mod.rs [hunk 2]
```diff
@@ -35,3 +37,48 @@ fn lower_string_from_env_var(arg_name: &str, string_from_env: &StringFromEnvVar)
         },
     }
 }
+
+fn primary_key_name_matches(pk: &PrimaryKeyDefinition, model: &Model, connector: &dyn Connector) -> bool {
```
Comment (tomhoule) [inline libs/datamodel/core/src/transform/dml_to_ast/mod.rs:41]: these just moved, logic is untouched

File: libs/datamodel/connectors/datamodel-connector/src/constraint_names.rs [hunk 3]
```diff
@@ -34,11 +31,7 @@ impl ConstraintNames {
     /// seq for sequences
     ///
 
-    pub fn primary_key_name_matches(pk: &PrimaryKeyDefinition, model: &Model, connector: &dyn Connector) -> bool {
```
Comment (pimeys) [inline libs/datamodel/connectors/datamodel-connector/src/constraint_names.rs:37]: 💯

File: libs/datamodel/connectors/datamodel-connector/src/constraint_names.rs [hunk 4]
```diff
@@ -116,32 +93,7 @@ impl ConstraintNames {
         format!("{}_df", trimmed)
     }
 
-    pub fn foreign_key_name_matches(ri: &RelationInfo, model: &Model, connector: &dyn Connector) -> bool {
```
Comment (pimeys) [inline libs/datamodel/connectors/datamodel-connector/src/constraint_names.rs:119]: 💯

File: libs/datamodel/connectors/sql-datamodel-connector/src/lib.rs [hunk 5]
```diff
@@ -8,12 +8,8 @@ mod sqlite_datamodel_connector;
 
 use datamodel_connector::Connector;
 
-pub struct SqlDatamodelConnectors;
```
Comment (pimeys) [inline libs/datamodel/connectors/sql-datamodel-connector/src/lib.rs:11]: 💯

**Review by pimeys — APPROVED** [review]

**Pull request merged**